File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        nums.sort()
        left = 0
        right = 1
        if len(nums)==1:
            return nums[0]
        while left<len(nums):
            if left == len(nums)-1:
                return nums[left]
            if nums[left] != nums[right]:
                return nums[left]
            else:
                left+=2
                right +=2
            
                

File: 0136-single-number/0136-single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        map<int,int> x;
        int answer;
        for(int i=0;i<nums.size();i++)
        {
            x[nums[i]]++;
        }
        for(auto &it : x) { 
               if(it.second == 1) { 
                  answer = it.first; 
                             } 
                           } 
        return answer;
    }
};